website/nifti-viewer/index.js

```diff
@@ -0,0 +1,232 @@
+    const stack = {
+        currentImageIdIndex: imageIdObject.slice.index,
+        imageIds: Array.from({ length: numberOfSlices }, (_, i) => `nifti:${imageIdObject.filePath}#${imageIdObject.slice.dimension}-${i},t-0`)
+    };
+
+    cornerstoneTools.addStackStateManager(element, ['stack']);
+    cornerstoneTools.addToolState(element, 'stack', stack);
+    cornerstoneTools.mouseInput.enable(element);
+    cornerstoneTools.mouseWheelInput.enable(element);
+    cornerstoneTools.pan.activate(element, 2);
+    cornerstoneTools.stackScrollWheel.activate(element);
+    cornerstoneTools.orientationMarkers.enable(element);
+    cornerstoneTools.stackPrefetch.enable(element);
+    cornerstoneTools.referenceLines.tool.enable(element, synchronizer);
+    cornerstoneTools.crosshairs.enable(element, 1, synchronizer);
+}
+
+// Handle voxel click event
+function handleVoxelClick(currentVoxel) {
+    const [nx, ny, nz, nt] = dims;
+    let [voxelX, voxelY, voxelZ] = currentVoxel;
+
+    voxelX = Math.min(Math.max(Math.round(voxelX), 1), nx) - 1;
+    voxelY = Math.min(Math.max(ny - Math.round(voxelY), 1), ny) - 1;
+    voxelZ = Math.min(Math.max(nz - Math.round(voxelZ), 1), nz) - 1;
+
+    const voxelValues = getVoxelValuesAcrossTime(voxelX, voxelY, voxelZ, nx, ny, nz, nt);
+    updateVoxelCoordinatesDisplay(voxelX + 1, voxelY + 1, voxelZ + 1);
+    plotVoxelData(voxelValues);
+}
+
+// Extract voxel values across all time points
+function getVoxelValuesAcrossTime(x, y, z, nx, ny, nz, nt) {
+    const sliceSize = nx * ny;
+    const volumeSize = sliceSize * nz;
+    let voxelValues = [];
+
+    for (let t = 0; t < nt; t++) {
+        const voxelIndex = x + y * nx + z * nx * ny + t * volumeSize;
+        voxelValues.push(niftiImageBuffer[voxelIndex]);
+    }
+
+    return voxelValues;
+}
+
+// Update voxel coordinates display on the page
+function updateVoxelCoordinatesDisplay(x, y, z) {
+    document.getElementById('voxel-coordinates').innerText = `(x, y, z): (${x}, ${y}, ${z})`;
+}
+
+// Update the slice index based on the current view
+function updateSliceIndex(view, newIndex) {
+    if (view === 'axial') {
+        voxel3dUnits[2] = newIndex;
+    } else if (view === 'sagittal') {
+        voxel3dUnits[0] = newIndex;
+    } else if (view === 'coronal') {
+        voxel3dUnits[1] = newIndex;
+    }
+    handleVoxelClick(voxel3dUnits);
+}
+
+// Load NIfTI file and display the axial, sagittal, and coronal views
+function loadAllFileViews(file) {
+    const fileURL = URL.createObjectURL(file);
+    const imageId = `nifti:${fileURL}`;
+
+    cornerstoneNIFTIImageLoader.nifti.loadHeader(imageId).then((header) => {
+        dims = [...header.voxelLength, header.timeSlices];
+        loadAndViewImage(document.getElementById('nifti-image-z'), `${imageId}#z,t-0`, 'axial');
+        loadAndViewImage(document.getElementById('nifti-image-x'), `${imageId}#x,t-0`, 'sagittal');
```

Comment:
There is a bug in synchronizing the sagittal view initially that I'm still trying to figure out.
For now you can fix this by scrolling through the view that is not synched, it will work normally after that.

Reply:
Actually I'm noticing it's both the sagittal and coronal views that don't act correctly.

Reply:
Same problem sharing images, sorry. Now I'm not seeing the coronal view misbehave. Definitely the sagittal one though. Sounds like you see that too though.